src/codegen/compile.ml

```diff
@@ -3478,7 +3478,10 @@ module IC = struct
 
     let pre_upgrade_fi = E.add_fun env "pre_upgrade" (Func.of_body env [] [] (fun env ->
       Lifecycle.trans env Lifecycle.InPreUpgrade ^^
-      G.i (Call (nr (E.built_in env "pre_exp"))) ^^
+      ClosureTable.count env ^^
+      G.if_ []
+       (E.trap_with env "canister_pre_upgrade trapped due to outstanding message callbacks (try stopping the canister before upgrade)")
```

Comment:
"trap" is already contained in the execution level string "Canister rwlgt-iiaaa-aaaaa-aaaaa-cai _trapped_ explicitly", so we shouldn't repeat it. Suggesting:
```suggestion
       (E.trap_with env "canister_pre_upgrade attempted with outstanding message callbacks (try stopping the canister before upgrade)")
```

Reply:
Done